kore/src/Kore/Repl/Interpreter.hs

```diff
@@ -399,20 +399,24 @@ printRewriteRule rule = do
 performSingleStep
     :: ReplM level StepResult
 performSingleStep = do
-    ReplState { claims , axioms , graph , claim , node, stepper } <- get
-    (graph'@Strategy.ExecutionGraph { graph = gr }, res) <- lift $ stepper claim claims axioms graph node
-    if res
-        then do
-            lensGraph .= graph'
-            let
-                context = Graph.context gr node
-            case Graph.suc' context of
-                [] -> pure NoChildNodes
-                [configNo] -> do
-                    lensNode .= configNo
-                    pure Success
-                neighbors -> pure (Branch neighbors)
-        else pure NodeAlreadyEvaluated
+    ReplState { claims , axioms , graph , claim , node , stepper } <- get
+    let
+        leaf = loop (loopCond . Strategy.graph $ graph) node
+    lensNode .= leaf
+    (graph'@Strategy.ExecutionGraph { graph = gr }, res) <- lift $ stepper claim claims axioms graph leaf
```

Comment:
If this line is longer than 80 chars, you can break it after the `<-` on the next line.